docs/spec/trustbloc-did-method.md

```diff
@@ -436,14 +454,15 @@ At this point, discovery and verification are complete. The client can now execu
   - Cache validated config files.
   - Present the list of endpoints parsed from said stakeholder configs, to be used for client DID method operations.
 
-### Error Cases
-Error cases which terminate the discovery process in a failure state:
-- Consortium config unavailable: The consortium domain points to a server that isn't functional
-- Insufficient endorsement: Insufficient stakeholders endorse the consortium
+## Security Considerations
+_This section is non-normative._
 
-Error cases for a specific stakeholder, which can be ignored if sufficient other stakeholders are available:
-- Stakeholder down: if a stakeholder's config servers are all down
-- Stakeholder endpoints unavailable: if none of the stakeholder's Sidetree endpoints are functional
-- No signature: The stakeholder has not signed a config it is expected to
-- Invalid stakeholder signature: if a stakeholder signature fails to verify against the stakeholder's verification key(s)
-- Inconsistent configuration: if the stakeholder mirrors consortium files, and these files are inconsistent across stakeholders.
+### Cryptographic Agility
+
+The TrustBloc DID Method relies on cryptographic hashes and signatures to prove stakeholder endorsement of consortium config files. The specific primitives used are identified in the configuration, so they can be replaced with different primitives at a later time - even within the same consortium. The history system maintains a traceable history of configuration changes, so the configuration can be verified to have a continuity of endorsement leading from the beginning of the consortium.
+
+This allows cryptographic primitives to be phased out and replaced as needed, while the consortium may remain in continuous use for discovery and DID method operations.
+
+### Trust on First Use
+
+The TrustBloc DID Method is designed to address the trust-on-first-use (TOFU) problem with respect to trusting DID method endpoints. The [bootstrapping](#bootstrapping-trust) section details several methods by which a client may trust and verify the configuration of a consortium, and from there, use a set of DID method operation endpoints (ie, Sidetree endpoints) which it can trust on the basis of their inclusion in the configurations of endorsed stakeholders.
```

Comment:
Can we add ##Privacy Considerations as well? Some ideas (and I don't think we need a lot here)
- although the public DID traceability problem still exists, consortiums will have different DIDs for entities which decreases their risk of observability and linkability.
- consortiums may have strict requirements on which clients an access a given DID registry, preventing DID resolution (or even consortium discovery) to unauthorized parties.
- DID Documents are not recorded on a permanent ledger, and therefore can be purged from consortium registries (while their existence still remains for forensic purposes?)

Anything else?

Reply:
Here are two generic ones from the DID Core spec.
https://www.w3.org/TR/did-core/#privacy-considerations

Keep Personally-Identifiable Information (PII) Private
Note that DID documents should not contain PII.

DID Correlation Risks and Pseudonymous DIDs
Note that public DIDs have risk of correlation and pairwise unique DIDs should be used where correlation is problematic. Also suggest did:key and did:peer as examples of pairwise identifiers.

Reply:
@mavarley 

> DID Documents are not recorded on a permanent ledger, and therefore can be purged from consortium registries (while their existence still remains for forensic purposes?)

It is true that the DID documents are not in the ledger. However, operations for multiple DIDs are mixed together in a Chunk file that is stored in the off-ledger storage. We should not delete the chunk file because that would impact the history of the other mixed-together DID operations. https://identity.foundation/sidetree/docs/spec/#chunk-files

Reply:
Anchor files are anchored to the target ledger system via embedding a CAS URI in the ledger’s transactional history. Once an Anchor File, Map File, and associated Chunk Files have been assembled for a given set of operations, a reference to the Anchor File (called Anchor String) is embedded within the target ledger.
Anchor string:
numOfOperations + “.” + CID of the anchor file

Reply:
If we need to re-process Sidetree transactions (or another party wants to do it) chunk files need to be available.

Reply:
https://github.com/trustbloc/trustbloc-did-method/issues/69